File: 2002B2/CS/Winforms-Controls-ListBoxItemData/ListBoxItemData/ListBoxItemData/Customer.cs
using System;
using System.IO;

namespace ListBoxItemData
{
	/// <summary>
	/// Summary description for Customer.
	/// </summary>
	public class Customer
	{
		private string strID;
		private string strTitle;
		private string strFirstName;
		private string strLastName;
		private string strAddress;
		private DateTime dtDateOfBirth;

		public Customer(string ID)
		{
			this.strID = ID;
			//
			// TODO: Add constructor logic here
			//
		}

		public static Customer[] LoadCustomers()
		{
			Customer[] Customers = new Customer[5];
			Customers[0] = ReadCustomer1();
			Customers[1] = ReadCustomer2();
			Customers[2] = ReadCustomer3();
			Customers[3] = ReadCustomer4();
			Customers[4] = ReadCustomer5();

			return Customers;

		}

		internal static Customer ReadCustomer1()
		{
			Customer cust = new Customer("536-45-1245");
			cust.Title = "Mr.";
			cust.FirstName = "Otis";
			cust.LastName = "Redding";
			cust.DateOfBirth = DateTime.Parse("9/9/1941");
			cust.Address = "1 Dock Street," + "\r\n" + "The Bay," + "\r\n" + "Georgia," + "\r\n" + "USA";
			return cust;
		}

		internal static Customer ReadCustomer2()
		{
			Customer cust = new Customer("246-12-5645");
			cust.Title = "Mr.";
			cust.FirstName = "James";
			cust.LastName = "Brown";
			cust.DateOfBirth = DateTime.Parse("5/3/1933");
			cust.Address = "45 New Bag Street," + "\r\n" + "Barnwell," + "\r\n" + "South Carolina," + "\r\n" + "USA";
			return cust;
		}

		internal static Customer ReadCustomer3()
		{
			Customer cust = new Customer("651-27-8117");
			cust.Title = "Mz.";
			cust.FirstName = "Aretha";
			cust.LastName = "Franklin";
			cust.DateOfBirth = DateTime.Parse("3/25/1942");
			cust.Address = "21 Chain Ave.," + "\r\n" + "Memphis," + "\r\n" + "Tennessee," + "\r\n" + "USA";
			return cust;
		}

		private static Customer ReadCustomer4()
		{
			Customer cust = new Customer("786-34-2114");
			cust.Title = "Mr.";
			cust.FirstName = "Louis";
			cust.LastName = "Armstrong";
			cust.DateOfBirth = DateTime.Parse("8/4/1901");
			cust.Address = "The West End," + "\r\n" + "New Orleans," + "\r\n" + "Lousiana," + "\r\n" + "USA";
			return cust;
		}

		private static Customer ReadCustomer5()
		{
			Customer cust = new Customer("445-34-4332");
			cust.Title = "Mz.";
			cust.FirstName = "Billie";
			cust.LastName = "Holiday";
			cust.DateOfBirth = DateTime.Parse("4/17/1915");
			cust.Address = "Southern Breeze Ave.," + "\r\n" + "\r\n" + "Baltimore," + "\r\n" + "Maryland," + "\r\n" + "USA";
			return cust;
		}

		public string ID
		//public readonly string ID
		{
			get
			{
				return strID;
			}
		}

		public string FirstName
		{
			get
			{
				return strFirstName;
			}
			set
			{
				strFirstName = value;
			}
		}

		public string Title
		{
			get
			{
				return strTitle;
			}
			set
			{
				strTitle = value;
			}
		}

		public string LastName
		{
			get
			{
				return strLastName;
			}
			set
			{
				strLastName = value;
			}
		}

		public string Address
		{
			get
			{
				return strAddress;
			}
			set
			{
				strAddress = value;
			}
		}

		public DateTime DateOfBirth
		{
			get
			{
				return dtDateOfBirth;
			}
			set
			{
				dtDateOfBirth = value;
			}
		}

		public override string ToString()
		{
			StringWriter sb = new StringWriter();
			sb.Write(this.Title);
			sb.Write(" ");
			sb.Write(this.FirstName);
			sb.Write(" ");
			sb.Write(this.LastName);
			return sb.ToString();
		}
	}
}

File: 2002B2/VB/Winforms-Controls-ListBoxItemData/ListBoxItemData/ListBoxItemData/Customer.vb
Imports System.IO

Public Class Customer
    
    Private mCustID As String
    Private mTitle As String
    Private mFirstname As String
    Private mLastName As String
    Private mAddress As String
    Private mDateOfBirth As DateTime
    
    Public Shared Function LoadCustomers() As Customer()
        
        Dim Customers(4) As Customer
        Customers(0) = ReadCustomer1()
        Customers(1) = ReadCustomer2()
        Customers(2) = ReadCustomer3()
        Customers(3) = ReadCustomer4()
        Customers(4) = ReadCustomer5()
        
        Return Customers
        
    End Function
    
    Friend Shared Function ReadCustomer1() As Customer
        Dim Cust As Customer = New Customer("536-45-1245")
        Cust.Title = "Mr."
        Cust.FirstName = "Otis"
        Cust.LastName = "Redding"
        Cust.DateOfBirth = DateTime.Parse("9/9/1941")
        Cust.Address = "1 Dock Street," + Strings.Chr(13) + Strings.Chr(10) + "The Bay," + Strings.Chr(13) + Strings.Chr(10) + "Georgia," + Strings.Chr(13) + Strings.Chr(10) + "USA"
        Return Cust
    End Function
    
    Friend Shared Function ReadCustomer2() As Customer
        Dim Cust As Customer = New Customer("246-12-5645")
        Cust.Title = "Mr."
        Cust.FirstName = "James"
        Cust.LastName = "Brown"
        Cust.DateOfBirth = DateTime.Parse("5/3/1933")
        Cust.Address = "45 New Bag Street," + Strings.Chr(13) + Strings.Chr(10) + "Barnwell," + Strings.Chr(13) + Strings.Chr(10) + "South Carolina," + Strings.Chr(13) + Strings.Chr(10) + "USA"
        Return Cust
    End Function
    
    Friend Shared Function ReadCustomer3() As Customer
        Dim Cust As Customer = New Customer("651-27-8117")
        Cust.Title = "Mz."
        Cust.FirstName = "Aretha"
        Cust.LastName = "Franklin"
        Cust.DateOfBirth = DateTime.Parse("3/25/1942")
        Cust.Address = "21 Chain Ave.," + Strings.Chr(13) + Strings.Chr(10) + "Memphis," + Strings.Chr(13) + Strings.Chr(10) + "Tennessee," + Strings.Chr(13) + Strings.Chr(10) + "USA"
        Return Cust
    End Function
    
    Private Shared Function ReadCustomer4() As Customer
        Dim Cust As Customer = New Customer("786-34-2114")
        Cust.Title = "Mr."
        Cust.FirstName = "Louis"
        Cust.LastName = "Armstrong"
        Cust.DateOfBirth = DateTime.Parse("8/4/1901")
        Cust.Address = "The West End," + Strings.Chr(13) + Strings.Chr(10) + "New Orleans," + Strings.Chr(13) + Strings.Chr(10) + "Lousiana," + Strings.Chr(13) + Strings.Chr(10) + "USA"
        Return Cust
    End Function
    
    Private Shared Function ReadCustomer5() As Customer
        Dim Cust As Customer = New Customer("445-34-4332")
        Cust.Title = "Mz."
        Cust.FirstName = "Billie"
        Cust.LastName = "Holiday"
        Cust.DateOfBirth = DateTime.Parse("4/17/1915")
        Cust.Address = "Southern Breeze Ave.," + Strings.Chr(13) + Strings.Chr(10) + "Baltimore," + Strings.Chr(13) + Strings.Chr(10) + "Maryland," + Strings.Chr(13) + Strings.Chr(10) + "USA"
        Return Cust
    End Function
    
    Public Sub New(ByVal ID As String)
        MyBase.New()
        mCustID = ID
    End Sub
    
    Public ReadOnly Property ID() As String
        Get
            Return mCustID
        End Get
    End Property
    
    Public Property FirstName() As String
        Get
            Return mFirstName
        End Get
        Set
            mFirstName = Value
        End Set
    End Property
    
    Public Property Title() As String
        Get
            Return mTitle
        End Get
        Set
            mTitle = Value
        End Set
    End Property
    
    Public Property LastName() As String
        Get
            Return mLastName
        End Get
        Set
            mLastName = Value
        End Set
    End Property
    
    Public Property Address() As String
        Get
            Return mAddress
        End Get
        Set
            mAddress = Value
        End Set
    End Property
    
    Public Property DateOfBirth() As DateTime
        Get
            Return mDateOfBirth
        End Get
        Set(ByVal Value As DateTime)
            mDateOfBirth = Value
        End Set
    End Property

    Public Overrides Function ToString() As String
        Dim sb As StringWriter = New StringWriter()
        sb.Write(Me.Title)
        sb.Write(" ")
        sb.Write(Me.FirstName)
        sb.Write(" ")
        sb.Write(Me.LastName)
        Return sb.ToString()
    End Function


End Class
